src/qvimagecore.cpp

```diff
@@ -232,7 +234,7 @@ QFileInfoList QVImageCore::getCompatibleFiles()
     const auto &regs = qvApp->getFilterRegExpList();
     const auto &mimeTypes = qvApp->getMimeTypeNameList();
 
-    const QFileInfoList currentFolder = currentFileDetails.fileInfo.dir().entryInfoList();
+    const QFileInfoList currentFolder = currentFileDetails.fileInfo.dir().entryInfoList(QDir::Filter::Files, QDir::SortFlag::Unsorted);
```

Comment:
Passing `QDir::SortFlag::Unsorted` forces it to skip the sort step, versus `QDir::SortFlag::NoSort` which ends up sorting it based on some default rules. Note that if you saw my previous commit where I accomplished something similar using `QDirIterator`, I ended up reverting that due to inconsistent performance when testing on a larger set of files.